### DISCUSSION TITLE
Gatsby V3 Incremental Builds - How do they work?

### DISCUSSION THREAD

**Discussion by hk86** [gatsbyjs/gatsby#31200] [Help]
Hi there,

I've built es website with Gatsby V3 and use Gatsby Cloud. The first build takes approx. 5 minutes. If I immediately after hit "Build", it takes again approx. 5 minutes to build. Incremental build seems not to work. The data is sourced from a mongodb. In gatsby-node.js I the use a graphql query and based on the query I create the pages. What do I miss? How the build process knows which pages need to be recreated?

Best regards,

HK

**Comment by KyleAMathews** [comment]
Where is time spent during your builds?

**Reply by hk86** [reply]
The time consuming tasks from the logs:
```
 16.35s - source and transform nodes
  7.38s - gatsby-source-wordpress pull updates since last build
128.12s - run page queries
 31.86s - Building production JavaScript and CSS bundles
  8.08s - Building HTML renderer
  8.60s - Building static HTML for pages
 11.24s - Uploading
 10.66s - Deploying Build to Netlify
```

**Comment by LekoArts** [comment]
You can read more here: https://www.gatsbyjs.com/docs/debugging-incremental-builds/

**Reply by hk86** [reply]
Thanks. This helps.
